ipa-macros/src/step/mod.rs

```diff
@@ -0,0 +1,32 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::parse_macro_input;
+
+/// Generate a state transition graph and the corresponding `StepNarrow` implementations for the
+/// IPA protocol.
+pub fn expand(_attr: TokenStream, input: TokenStream) -> TokenStream {
```

Comment:
as we are basically implementing a trait, I believe `Derive(step)` is what we would want?

Reply:
I'm using attribute proc macro here because of two features that are not available in derive proc macro. 1) it replaces itself with any tokens during macro expansion so it allows me to add a derive trait, and 2) supports attributes that are applied to the entire item such as `#[step(obsolete)]`. I have a new diff that this attribute macro adds `derive(Step)`

Reply:
yea, we should use `derive(Step)` and let end users derive other attributes, even if they are common for all steps.